datalab/datalab_session/data_operations/median.py

```diff
@@ -42,63 +40,40 @@ def wizard_description():
             }
         }
     
-    def operate(self):
-        input_files = self.input_data.get('input_files', [])
-        file_count = len(input_files)
+    def operate(self, input_files, cache_key):
 
-        if file_count == 0:
-            return { 'output_files': [] }
+        log.info(f'Executing median operation on {len(input_files)} files')
 
-        log.info(f'Executing median operation on {file_count} files')
+        image_data_list = load_image_data_from_fits_urls(input_files)
 
-        with tempfile.TemporaryDirectory() as temp_dir:
-            memmap_paths = []
+        self.set_percent_completion(0.4)
 
-            for index, file_info in enumerate(input_files):
-                basename = file_info.get('basename', 'No basename found')
-                archive_record = get_archive_from_basename(basename)
+        stacked_data = stack_arrays(image_data_list)
 
-                try:
-                    fits_url = archive_record[0].get('url', 'No URL found')
-                except IndexError:
-                    continue
+        median = np.median(stacked_data, axis=2)
 
-                with fits.open(fits_url, use_fsspec=True) as hdu_list:
-                    data = hdu_list['SCI'].data
-                    memmap_path = os.path.join(temp_dir, f'memmap_{index}.dat')
-                    memmap_array = np.memmap(memmap_path, dtype=data.dtype, mode='w+', shape=data.shape)
-                    memmap_array[:] = data[:]
-                    memmap_paths.append(memmap_path)
+        hdu_list = create_fits(cache_key, median)
 
-                self.set_percent_completion(index / file_count)
+        # Create the output files to be stored in S3
```

Comment:
I think it would probably be worth pulling this into a function in the base DataOperation for others to use, that takes in a first image and creates a small and large thumbnail and uploads them all to the s3 bucket, and returns the url to the files as a result. A lot of this code should be re-usable for many other operations that have fits files as output.